_search-plugins/search-pipelines/debugging-search-pipeline.md

```diff
@@ -0,0 +1,175 @@
+
+- [Specifying an existing pipeline for a request]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/using-search-pipeline/#specifying-an-existing-search-pipeline-for-a-request).
+
+- [Using a temporary pipeline for a request]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/using-search-pipeline/#using-a-temporary-search-pipeline-for-a-request).
+
+- [Setting a default pipeline for all requests in an index]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/using-search-pipeline/#default-search-pipeline).
+
+  To enable the `verbose_pipeline` feature, simply add `verbose_pipeline=true` as a query parameter in your search request.
+
+### Example request with verbose pipeline
+
+You can specify the `verbose_pipeline` parameter in the query to enable detailed debugging. The `verbose_pipeline` parameter works seamlessly across all search pipeline configurations:
+
+#### Default Search Pipeline
+
+To use `verbose_pipeline` with a default search pipeline, set the pipeline as the default in the index settings and include `verbose_pipeline=true` in the query:
+
+```json
+PUT /my_index/_settings
+{
+  "index.search.default_pipeline": "my_pipeline"
+}
+```
+{% include copy-curl.html %}
+
+```json
+GET /my_index/_search?verbose_pipeline=true
+```
+{% include copy-curl.html %}
+
+#### Specified Search Pipeline by ID
+To use `verbose_pipeline` with a specific search pipeline, specify the pipeline ID along with verbose_pipeline=true in the query:
+```json
+GET /my_index/_search?search_pipeline=my_pipeline&verbose_pipeline=true
+```
+{% include copy-curl.html %}
+
+#### Temporary Search Pipeline
+To use `verbose_pipeline` with a temporary search pipeline, define the pipeline directly in the request body and include verbose_pipeline=true in the query:
+```json
+POST /my_index/_search?verbose_pipeline=true
+{
+  "query": {
+    "match": { "text_field": "some search text" }
+  },
+  "search_pipeline": {
+    "request_processors": [
+      {
+        "filter_query": {
+          "query": { "term": { "visibility": "public" } }
+        }
+      }
+    ],
+    "response_processors": [
+      {
+        "collapse": {
+          "field": "category"
+        }
+      }
+    ]
+  }
+}
+```
+{% include copy-curl.html %}
+### Example response with verbose pipeline
+
+When the `verbose_pipeline` parameter is enabled, the response contains an additional `processor_results` field that details the transformations applied by each processor in the pipeline:
+
+<details open markdown="block">
+  <summary>
+    Response
+  </summary>
+  {: .text-delta}
+
+```json
+{
+  "took": 40,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.0,
+    "hits": [
+      {
+        "_index": "my_index",
+        "_id": "1",
+        "_score": 0.0,
```

Comment:
can you make the score something positive, even if it's just for the sake of example. matches with score 0.0 looks a bit suspicious.